### ISSUE TITLE
Aborted (core dumped) during running raxml-ng

### ISSUE THREAD

**Issue by TakashiKoyama** [amkozlov/raxml-ng#121]
Hi,

I tried to raxml-ng with my phy input generated from VCF but got error as below.
How can I resolve this?
```
$ raxml-ng --all --msa Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy --model GTR+G+ASC_LEWIS --bs-trees 200 --seed 2 --threads 40 --prefix bs200seed2

RAxML-NG v. 1.0.2 released on 22.02.2021 by The Exelixis Lab.
Developed by: Alexey M. Kozlov and Alexandros Stamatakis.
Contributors: Diego Darriba, Tomas Flouri, Benoit Morel, Sarah Lutteropp, Ben Bettisworth.
Latest version: https://github.com/amkozlov/raxml-ng
Questions/problems/suggestions? Please visit: https://groups.google.com/forum/#!forum/raxml

System: Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz, 20 cores, 125 GB RAM

RAxML-NG was called at 09-Jul-2021 17:35:50 as follows:

raxml-ng --all --msa Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy --model GTR+G+ASC_LEWIS --bs-trees 200 --seed 2 --threads 40 --prefix bs200seed2

Analysis options:
  run mode: ML tree search + bootstrapping (Felsenstein Bootstrap)
  start tree(s): random (10) + parsimony (10)
  bootstrap replicates: 200
  random seed: 2
  tip-inner: OFF
  pattern compression: ON
  per-rate scalers: OFF
  site repeats: ON
  branch lengths: proportional (ML estimate, algorithm: NR-FAST)
  SIMD kernels: AVX
  parallelization: coarse-grained (auto), PTHREADS (40 threads), thread pinning: OFF

[00:00:00] Reading alignment from file: Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy
[00:00:00] Loaded alignment with 38 taxa and 1466137 sites

Alignment comprises 1 partitions and 159000 patterns

Partition 0: noname
Model: GTR+FO+G4m+ASC_LEWIS
Alignment sites / patterns: 1466137 / 159000
Gaps: 0.00 %
Invariant sites: 0.00 %


NOTE: Binary MSA file created: bs200seed2.raxml.rba

Parallelization scheme autoconfig: 2 worker(s) x 20 thread(s)

Parallel reduction/worker buffer size: 1 KB  / 0 KB

[00:00:01] Generating 10 random starting tree(s) with 38 taxa
[00:00:01] Generating 10 parsimony starting tree(s) with 38 taxa
[00:00:08] Data distribution: max. partitions/sites/weight per thread: 1 / 7950 / 127200
[00:00:19] Data distribution: max. searches per worker: 110

Starting ML tree search with 20 distinct starting trees

LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
Aborted (core dumped)
```

**Comment by amkozlov** [comment]
Hello Takashi, could you please send me your input file (`.phy`), then I will try to reproduce this bug.

**Comment by TakashiKoyama** [comment]
Thank you for your reply.
I attached my .phy file.
[Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy.zip](https://github.com/amkozlov/raxml-ng/files/6798187/Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy.zip)

**Comment by amkozlov** [comment]
Hi @TakashiKoyama, so turns out this problem is caused by the columns which only differ in ambiguos characters, e.g.

```
A
A
A
R
```
For the purpose of asc. bias correction, such columns are also considered invariant and thus have to be excluded for mathematical reasons.

Unfortunately, raxml-ng currently does not detect such cases, but I will fix it.

**Comment by TakashiKoyama** [comment]
Thank you for elucidating this problem!
Could you suggest any tool to exclude such sites?

**Comment by amkozlov** [comment]
maybe try this one:
https://github.com/btmartin721/raxml_ascbias

**Comment by TakashiKoyama** [comment]
Just for your information.

I have tried ascbias.py.
```
$ ascbias.py -p Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy -o Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.WOascbias.phy
```
Files should be different each other (but I am not sure sites you suggested were excluded).
```
$ ls -alth *.phy
-rw-rw-r--. 1 koyama koyama 31M Jul 19 20:38 Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.ascbias.phy
-rw-rw-r--. 1 koyama koyama 54M Jul  1 22:30 Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.phy
```
But got same error in raxml-ng.
```
$ raxml-ng --all --msa Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.ascbias.phy --model GTR+G+ASC_LEWIS --bs-trees 200 --seed 2 --threads 40 --prefix bs200seed2

RAxML-NG v. 1.0.2 released on 22.02.2021 by The Exelixis Lab.
Developed by: Alexey M. Kozlov and Alexandros Stamatakis.
Contributors: Diego Darriba, Tomas Flouri, Benoit Morel, Sarah Lutteropp, Ben Bettisworth.
Latest version: https://github.com/amkozlov/raxml-ng
Questions/problems/suggestions? Please visit: https://groups.google.com/forum/#!forum/raxml

System: Intel(R) Xeon(R) CPU E5-2690 v2 @ 3.00GHz, 20 cores, 125 GB RAM

RAxML-NG was called at 19-Jul-2021 21:25:37 as follows:

raxml-ng --all --msa Seriola.BQSR3_MaximumLikelihood.merged.autosomes.VS.smple.2k50k.min38.ascbias.phy --model GTR+G+ASC_LEWIS --bs-trees 200 --seed 2 --threads 40 --prefix bs200seed2

Analysis options:
  run mode: ML tree search + bootstrapping (Felsenstein Bootstrap)
  start tree(s): random (10) + parsimony (10)
  bootstrap replicates: 200
  random seed: 2
  tip-inner: OFF
  pattern compression: ON
  per-rate scalers: OFF
  site repeats: ON
  branch lengths: proportional (ML estimate, algorithm: NR-FAST)
  SIMD kernels: AVX
  parallelization: coarse-grained (auto), PTHREADS (40 threads), thread pinning: OFF

WARNING: The model you specified on the command line (GTR+G+ASC_LEWIS) will be ignored 
         since the binary MSA file already contains a model definition.
         If you want to change the model, please re-run RAxML-NG 
         with the original PHYLIP/FASTA alignment and --redo option.

[00:00:00] Loading binary alignment from file: bs200seed2.raxml.rba
[00:00:00] Alignment comprises 38 taxa, 1 partitions and 159000 patterns

Partition 0: noname
Model: GTR+FO+G4m+ASC_LEWIS
Alignment sites / patterns: 1466137 / 159000
Gaps: 0.00 %
Invariant sites: 0.00 %


Parallelization scheme autoconfig: 2 worker(s) x 20 thread(s)

Parallel reduction/worker buffer size: 1 KB  / 0 KB

[00:00:00] NOTE: Resuming execution from checkpoint (logLH: -3872569.09, ML trees: 0, bootstraps: 0)
[00:00:00] Data distribution: max. partitions/sites/weight per thread: 1 / 7950 / 127200
[00:00:09] Data distribution: max. searches per worker: 110

Starting ML tree search with 20 distinct starting trees

LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
raxml-ng: /home/alex/hits/raxng-release/libs/pll-modules/src/algorithm/algo_search.c:1328: pllmod_algo_spr_round: Assertion `fabs(loglh - best_lh) < 1e-6' failed.
LH mismatch: -219735.264874355285  != -219735.264872521861
Aborted (core dumped)
```
I will try to find another way.

**Comment by amkozlov** [comment]
it is not using the new file, please add `--redo` option!

**Comment by TakashiKoyama** [comment]
ah, thank you!
Finally, it seems to be working.
Again, thank you for your advices!

**Comment by amkozlov** [comment]
fixed in v1.0.3
